File: Linked List/LeetCode_206 Reverse Linked List/Reverse Linked List.py
class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        cur, prev = head, None  
        while cur:
            cur.next, prev, cur = prev, cur, cur.next  #
        return prev
		
		
class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        if(not head or not head.next):
            return head
        node = self.reverseList(head.next)
        head.next.next = head
        head.next = None
        return node
        

File: Linked List/LeetCode_206 Reverse Linked List/Reverse Linked List.c
struct ListNode* reverseList(struct ListNode* head){
    struct ListNode *temp;
    struct ListNode *pPre,*pCur;
    pCur = head;
    pPre = NULL;
    while(pCur !=NULL){
        temp = pCur->next;
        pCur->next = pPre;
        pPre = pCur;
        pCur = temp;
    }
    return pPre;
}


struct ListNode* reverseList(struct ListNode* head){
    if (!head || !(head -> next)) {
            return head;
        }
    struct ListNode* node = reverseList(head -> next);  
    head -> next -> next = head;
    head -> next = NULL;  /* Tail pointing NULL*/
    return node;
}
